In [17]:
import sqlite3
import csv

# Path to your CSV file
csv_file_path = 'dataset/MCH_Indicators_2016_2022.csv'

# Replace 'your_database_name.db' with your desired file name
db_path = 'Maternal_Indicators.db'


# Connect to the SQLite database
conn = sqlite3.connect(db_path)

# Read CSV into DataFrame
df = pd.read_csv(csv_file_path)

# Write DataFrame to SQL Table, assume table name as 'new_table'
df.to_sql('new_table', con=conn, if_exists='replace', index=False)

# Close the connection
conn.close()

In [18]:
%sql sqlite:///Maternal_Indicators.db

In [19]:
%sql SELECT * FROM new_table LIMIT 10;

 * sqlite:///Maternal_Indicators.db
   sqlite:///your_database_name.db
Done.


[('Alaska', 2016, None, 1165, 435, 40.5, 37.0, 44.0, 1144, 34, 2.3, 1.5, 3.5, 1144, 267, 23.4, 20.5, 26.6, 1144, 289, 24.6, 21.7, 27.8, 1158, 330, 25.7, 22.9, 28.8, 1160, 162, 11.0, 9.2, 13.1, 1158, 213, 14.8, 12.7, 17.1, 1156, 50, 5.3, 3.8, 7.3, 1156, 15, 1.1, 0.6, 2.1, 1147, 39, 4.0, 2.7, 5.9, 1152, 43, 2.8, 1.9, 4.1, 1148, 50, 2.7, 1.9, 3.9, 1142, 39, 2.2, 1.5, 3.4, 1172, 160, 12.0, 10.0, 14.4, 1165, 181, 13.2, 11.1, 15.8, 1128, 155, 14.1, 11.7, 16.8, 1163, 724, 65.4, 62.1, 68.6, 1142, 964, 84.7, 81.9, 87.1, 1143, 696, 57.6, 53.9, 61.1, 1137, 968, 86.4, 83.9, 88.6, 1153, 190, 17.2, 14.6, 20.2, 1153, 59, 4.8, 3.5, 6.5, 1153, 212, 16.5, 14.1, 19.1, 1153, 692, 61.5, 58.0, 64.9, 1118, 872, 79.9, 76.9, 82.6, 1118, 135, 12.6, 10.3, 15.3, 1118, 278, 23.8, 20.9, 27.1, 1118, 230, 20.4, 17.6, 23.5, 1118, 229, 23.0, 20.0, 26.3, 1166, 521, 47.7, 44.2, 51.3, 1143, 611, 59.1, 55.8, 62.3, 1143, 304, 23.2, 20.6, 26.1, 1143, 228, 17.7, 15.2, 20.4),
 ('Arkansas', 2016, 17.0, 901, 298, 35.9, 31.4, 40.7, 864, 42, 3.8, 2.3, 6.0, 864, 203, 25.6, 21.5, 30.1, 864, 270, 28.9, 24.7, 33.6, 878, 246, 24.3, 20.4, 28.7, 880, 155, 15.7, 12.5, 19.6, 878, 201, 19.9, 16.3, 24.1, 884, 61, 8.3, 5.9, 11.6, 883, 28, 3.4, 2.0, 5.8, 883, 40, 4.4, 2.8, 7.0, 884, 19, 3.2, 1.8, 5.5, 880, 52, 6.0, 4.0, 8.7, 881, 28, 2.1, 1.1, 4.0, 900, 170, 19.0, 15.5, 23.1, 880, 187, 20.8, 17.1, 25.0, 884, 205, 19.2, 15.7, 23.3, 895, 491, 56.4, 51.5, 61.2, 886, 698, 80.1, 75.9, 83.8, 888, 483, 61.0, 56.2, 65.7, 875, 754, 89.9, 86.6, 92.5, 894, 198, 21.5, 17.7, 25.8, 894, 114, 9.8, 7.2, 13.0, 894, 190, 17.7, 14.3, 21.6, 894, 392, 51.1, 46.2, 56.0, 872, 704, 83.4, 79.4, 86.7, 872, 155, 14.7, 11.6, 18.5, 872, 110, 15.5, 12.2, 19.5, 872, 299, 34.8, 30.2, 39.7, 872, 140, 18.4, 14.8, 22.6, 890, 283, 36.2, 31.6, 41.1, 785, 468, 64.9, 59.7, 69.7, 785, 247, 24.2, 20.0, 28.9, 785, 70, 11.0, 8.0, 14.8),
 ('Colorado', 2016, 16.0, 1619, 753, 43.2, 39.9, 46.7, 1528, 63, 3.4, 2.4, 4.9, 1528, 325, 23.9, 21.0, 27.1, 1528, 274, 17.3, 14.7, 20.2, 1617, 262, 14.7, 12.4, 17.2, 1618, 116, 6.4, 5.0, 8.3, 1617, 175, 10.0, 8.1, 12.2, 1610, 61, 2.7, 1.8, 3.9, 1610, 17, 1.1, 0.6, 2.3, 1585, 77, 5.4, 4.1, 7.2, 1614, 74, 4.3, 3.2, 5.8, 1584, 40, 2.8, 1.8, 4.5, 1586, 29, 2.1, 1.2, 3.7, 1606, 208, 12.1, 10.0, 14.5, 1603, 208, 12.1, 10.0, 14.4, 1606, 185, 9.9, 8.0, 12.1, 1602, 1179, 69.0, 65.6, 72.3, 1592, 1425, 90.4, 88.0, 92.4, 1604, 1058, 66.7, 63.4, 69.8, 1611, 1476, 91.9, 89.7, 93.6, 1600, 271, 18.5, 15.9, 21.4, 1600, 68, 4.3, 3.0, 6.2, 1600, 234, 16.0, 13.5, 18.8, 1600, 1027, 61.2, 57.7, 64.6, 1586, 1258, 77.9, 74.8, 80.7, 1586, 193, 9.9, 8.1, 11.9, 1586, 352, 22.6, 19.8, 25.7, 1586, 347, 21.6, 18.8, 24.6, 1586, 366, 23.9, 21.0, 27.0, 1620, 768, 46.8, 43.4, 50.3, 1538, 1042, 66.3, 62.7, 69.6, 1538, 359, 24.1, 21.1, 27.3, 1538, 132, 9.5, 7.5, 12.1),
 ('Connecticut', 2016, None, 1286, 543, 47.8, 44.5, 51.1, 1232, 46, 3.6, 2.5, 5.1, 1232, 310, 24.6, 21.7, 27.7, 1232, 320, 26.1, 23.2, 29.3, 1269, 164, 16.5, 13.9, 19.4, 1271, 59, 5.5, 4.1, 7.5, 1271, 106, 9.4, 7.5, 11.7, 1276, 28, 2.6, 1.7, 4.0, 1280, 5, 0.3, 0.1, 0.7, 1274, 75, 5.1, 3.9, 6.6, 1272, 20, 2.1, 1.3, 3.6, 1278, 31, 2.2, 1.4, 3.4, 1275, 30, 2.3, 1.5, 3.7, 1284, 104, 8.9, 7.1, 11.1, 1266, 102, 8.7, 6.9, 10.9, 1265, 155, 10.5, 8.6, 12.6, 1280, 931, 78.7, 76.2, 81.0, 1272, 1150, 92.1, 90.3, 93.6, 1272, 882, 70.4, 67.2, 73.4, 1269, 1199, 95.4, 93.9, 96.5, 1279, 237, 16.9, 14.7, 19.5, 1279, 75, 4.7, 3.6, 6.2, 1279, 204, 14.6, 12.4, 17.2, 1279, 763, 63.7, 60.5, 66.8, 1256, 943, 75.2, 72.0, 78.1, 1256, 121, 9.5, 7.7, 11.7, 1256, 209, 16.9, 14.5, 19.6, 1256, 303, 25.1, 22.2, 28.2, 1256, 310, 23.7, 20.9, 26.7, 1284, 672, 58.8, 55.5, 62.0, 1263, 797, 67.4, 64.5, 70.1, 1263, 317, 23.2, 20.6, 26.0, 1263, 149, 9.4, 8.0, 11.1),
 ('Delaware', 2016, None, 928, 405, 41.8, 38.5, 45.1, 876, 29, 2.7, 1.8, 4.0, 876, 231, 26.9, 23.9, 30.1, 876, 237, 26.9, 24.0, 30.1, 923, 188, 20.1, 17.5, 22.9, 925, 95, 9.0, 7.3, 11.0, 925, 131, 13.0, 10.9, 15.5, 921, 34, 3.7, 2.7, 5.3, 92

## Dataset Overview

This analysis utilizes a dataset containing maternal health indicators across various US states from the years 2016 to 2021. Below is a breakdown of the columns present in the dataset and their significance:

- **Site Name**: Represents the US state where the data was recorded.
- **Year**: Indicates the year in which the data was collected, ranging from 2016 to 2021.
- **Deaths/100000**: The maternal mortality rate per 100,000 live births in the state for the given year.
- **Multivitamin use ≥4 times a week during the month before pregnancy - Unweighted Sample Size**: The number of survey respondents.
- **Multivitamin use ≥4 times a week during the month before pregnancy - Unweighted Frequency**: The raw count of respondents who reported using multivitamins more than four times a week.
- **Multivitamin use ≥4 times a week during the month before pregnancy - Weighted %**: The percentage of respondents, adjusted for the population, who used multivitamins as described.
- **Multivitamin use ≥4 times a week during the month before pregnancy - Lower 95% Confidence Interval**: The lower bound of the 95% confidence interval for the percentage.
- **Multivitamin use ≥4 times a week during the month before pregnancy - Upper 95% Confidence Interval**: The upper bound of the 95% confidence interval for the percentage.
- **Underweight (body mass index [BMI] <18.5 kg/m2) - Unweighted Sample Size**: The number of respondents classified as underweight.
- **Underweight (body mass index [BMI] <18.5 kg/m2)**: The percentage of respondents classified as underweight.

This dataset provides a comprehensive view of maternal health trends and pre-pregnancy behaviors, including nutritional supplementation and body weight, across different states and over a series of years. The following analysis aims to uncover patterns, correlations, and insights that could inform public health strategies and interventions.


## Data Cleaning
Before diving into the analysis, we need to ensure our dataset is clean and usable. This involves checking for missing values, correcting data types, and validating the consistency of the data across all years and states.


In [ ]:
%%sql 

## Descriptive Statistics
We will begin our analysis by computing basic statistics for each indicator. This includes averages, standard deviations, and ranges for maternal deaths, multivitamin usage, and BMI categories. This step helps us understand the general landscape of our data.


## Exploratory Data Analysis
Here, we will conduct a deeper exploration of the data. We aim to identify trends over the years and differences among states. Key focuses will be on changes in maternal death rates and how frequently multivitamins are used before pregnancy.


## In-depth Analysis
We will analyze the correlation between multivitamin use and maternal health outcomes. Additionally, we will investigate how maternal health indicators correlate with BMI categories. This section will involve more complex SQL queries and possibly statistical tests to understand these relationships.


## Comparative Analysis
In this section, we will compare maternal health indicators across different states. We will identify states with the best and worst health outcomes and explore possible reasons for these disparities.


## Conclusions
Based on our findings, we will summarize the key insights and suggest implications for public health policies. We will also recommend areas for further research based on data gaps or unexpected findings.
